tests/chgres_cube/ftst_sfc_input_data.F90

```diff
@@ -0,0 +1,94 @@
+ program test_sfc_input_data
+
+! Test sfc_input_data. For now only one test is performed
+! on the subroutine soilt_check using a sample 3x3 matrix.
+!
+! author: Larissa Reames (larissa.reames@noaa.gov)
+
+ use mpi
+ use esmf
+ use input_data
+ use model_grid
+
+ implicit none
+
+ integer :: ierr
+ 
+ integer(esmf_kind_i4), allocatable :: mask(:,:)
+ real(esmf_kind_r8), allocatable    :: soilt_bad(:,:,:), &
+                                       soilt_updated(:,:,:), &
+                                       soilt_correct(:,:,:)
+ real(esmf_kind_r8), allocatable    :: skint(:,:)
+
+ call mpi_init(ierr)
+
+ lsoil_input = 2
+ i_input = 3
+ j_input = 3
+
+ allocate(mask(i_input,j_input))
+ allocate(skint(i_input,j_input))
+ allocate(soilt_bad(i_input,j_input,lsoil_input))
+ allocate(soilt_updated(i_input,j_input,lsoil_input))
+ allocate(soilt_correct(i_input,j_input,lsoil_input))
+
+!--------------------------------------------------------
+! These variables are used to test all three functions of
+! the check_soilt routine (i.e., replace water point soil 
+! temperature and excessive land point soil temperature 
+! with skin temperature and store a default ice column
+! temperature in the soil temperature array since this
+! array isn't available in grib2 files)
+!--------------------------------------------------------
+
+! Definition of the mask. The '1' indicates an isolated
```

Comment:
In this case, 0 is open water, 1 is land, and 2 is ice.

Reply:
I've updated the comments in the most recent commit to address this and your previous comments.